test/functional/interface_rest.py

```diff
@@ -4,351 +4,297 @@
+        assert_greater_than_or_equal(int(hex_response.getheader('content-length')),
+                                     json_obj['size']*2)
 
+        self.log.info("Test tx inclusion in the /mempool and /block URIs")
 
-        # check block tx details
-        # let's make 3 tx and mine them on node 1
+        # Make 3 tx and mine them on node 1
         txs = []
-        txs.append(self.nodes[0].sendtoaddress(self.nodes[2].getnewaddress(), 11))
-        txs.append(self.nodes[0].sendtoaddress(self.nodes[2].getnewaddress(), 11))
-        txs.append(self.nodes[0].sendtoaddress(self.nodes[2].getnewaddress(), 11))
+        txs.append(self.nodes[0].sendtoaddress(not_related_address, 11))
+        txs.append(self.nodes[0].sendtoaddress(not_related_address, 11))
+        txs.append(self.nodes[0].sendtoaddress(not_related_address, 11))
         self.sync_all()
 
-        # check that there are exactly 3 transactions in the TX memory pool before generating the block
-        json_string = http_get_call(url.hostname, url.port, '/rest/mempool/info'+self.FORMAT_SEPARATOR+'json')
-        json_obj = json.loads(json_string)
+        # Check that there are exactly 3 transactions in the TX memory pool before generating the block
+        json_obj = self.test_rest_request("/mempool/info")
         assert_equal(json_obj['size'], 3)
         # the size of the memory pool should be greater than 3x ~100 bytes
         assert_greater_than(json_obj['bytes'], 300)
 
-        # check that there are our submitted transactions in the TX memory pool
-        json_string = http_get_call(url.hostname, url.port, '/rest/mempool/contents'+self.FORMAT_SEPARATOR+'json')
-        json_obj = json.loads(json_string)
+        # Check that there are our submitted transactions in the TX memory pool
+        json_obj = self.test_rest_request("/mempool/contents")
         for i, tx in enumerate(txs):
-            assert_equal(tx in json_obj, True)
-            assert_equal(json_obj[tx]['spentby'], txs[i+1:i+2])
-            assert_equal(json_obj[tx]['depends'], txs[i-1:i])
+            assert tx in json_obj
+            assert_equal(json_obj[tx]['spentby'], txs[i + 1:i + 2])
+            assert_equal(json_obj[tx]['depends'], txs[i - 1:i])
 
-        # now mine the transactions
+        # Now mine the transactions
         newblockhash = self.nodes[1].generate(1)
         self.sync_all()
 
-        #check if the 3 tx show up in the new block
-        json_string = http_get_call(url.hostname, url.port, '/rest/block/'+newblockhash[0]+self.FORMAT_SEPARATOR+'json')
-        json_obj = json.loads(json_string)
-        for tx in json_obj['tx']:
-            if not 'coinbase' in tx['vin'][0]: #exclude coinbase
-                assert_equal(tx['txid'] in txs, True)
+        # Check if the 3 tx show up in the new block
+        json_obj = self.test_rest_request("/block/{}".format(newblockhash[0]))
+        non_coinbase_txs = [tx['txid'] for tx in json_obj['tx']
```

Comment:
I'm not sure if ordering is guaranteed here. Perhaps construct sets and check for set equality?

Reply:
Sounds good - done.